File: Data Structures & Algorithms/jump-game/submission-3.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        maxi = nums[0]
        i = 0
        while i <= maxi:
            maxi = max(maxi, i + nums[i])
            i+=1
            if maxi >= len(nums) - 1:
                return True
        return False

File: Data Structures & Algorithms/eating-bananas/submission-3.cpp
class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        int maxN = 0;
        
        for(int p: piles){
            maxN = max(maxN, p);
        }
        int l = 1, r = maxN; 
        int minNeedHour = maxN;
        while(l <= r){
            int mid = (r+l) / 2;
            int totalNeed = 0;
            for(int i=0; i<piles.size(); i++){
                double needHour = piles[i];
                int need = ceil(needHour/mid);
                totalNeed += need;
            }

            if(totalNeed <= h){
                minNeedHour = min(minNeedHour, mid);
                r = mid - 1;
            }else{
                l = mid + 1;
            }

        }
        return minNeedHour;
    }
};
